code/modules/mob/living/living_defense.dm

```diff
@@ -201,6 +187,24 @@
 	adjust_fire_stacks(3)
 	IgniteMob()
 
+//Share fire evenly between the two mobs
+//Called in MobBump() and Crossed()
+/mob/living/proc/spreadFire(mob/living/L)

```

Comment:
Hi, I wrote this.
It's quite simple, I'll go through some examples:

A is on fire with 10 stacks, B is not on fire, with 0 stacks
A bumps into B and now they both have 5 stacks, and are both on fire (half of A's stacks went onto B, this is to represent the like, welding fuel/burning fuel etc. spreading on touch)

If they're both on fire (with 10 again) it's a little weirder, whoever initiated the bump (ie: someone potentially trying to use it as an attack) will end up with 12.5 and the victim will end up with 7.5.
(A's 10/2 = 5, B is now on 15, then B's stacks are split as B is also on fire, to 7.5, which is added back on to A, taking A to 12.5 and B to 7.5)

Amusingly, if both are on fire, and the initiator's stacks are exactly double the victim's, then nothing happens (half pass over from A to B, which takes B's value up to A's old value, and the same then happens again in reverse)

If you're both on fire and you have less stacks than them, you get saddled with more than them (taking A 5, B 10 as an example, A is left with 8.75 and B with 6.25)

The basic idea of it is: you can use this to help yourself, but a lot of the time it's just going to mess you up and cause chaos, which is the main goal.

Hope that explains it.